lib/router/router.js

```diff
@@ -224,6 +224,10 @@ export default class Router extends EventEmitter {
   }
 
   async prefetch (url) {
+    // We don't add support for prefetch in the development mode.
+    // If we do that, our on-demand-entries optimization won't performs better
+    if (process.env.NODE_ENV === 'development') return
```

Comment:
Some other places test with `process.env.NODE_ENV !== 'production'` instead.

Developers might have forgotten to define `process.env.NODE_ENV` but still expect to have a development behaviour in that case.

I think it would be nice if all the code tested whether it's in a production mode or not, in a coherent way.